### PR TITLE
Cypress/E2E: Fix CI run of Cypress and add flag to sort specs to test first

### PR DESCRIPTION
**Pull request by saturninoabril**
#### Summary
- Changes in `e2e/run_test_cycle.js` fix the issue where the container is being rerun unexpectedly. Probably due to mishandling of recursion thru async/await operation.
- In `e2e/utils/file.js` - added ability to sort specs to first. This is in the case for Elasticsearch where it's better to test first, so that indexing will not take a lot of time instead of doing last where there are lots of test data already. Typical used would be:
```
node generate_test_cycle.js --stage="@prod" --sortFirst="@elasticsearch,@ldap_group,@ldap" --sortLast="@saml,@keycloak,@mfa"
```

Test report - https://mm-cypress-report.s3.amazonaws.com/4454-master-e20-prod-mattermost-enterprise-edition:master/mochawesome.html

### CODE REVIEW

File: e2e/utils/file.js [hunk 1]
```diff
@@ -90,42 +90,60 @@ function getTestFiles() {
...
+function getWeightedFiles(metadata, sortFirst = true) {
+    let weightedFiles = [];
+    if (metadata) {
+        metadata.split(',').forEach((word, i, arr) => {
             const sl = grepCommand(word);
             const files = grepFiles(sl).map((file) => {
-                return {file, sortWeight: i + 1};
+                return {
+                    file,
+                    sortWeight: sortFirst ? (i - arr.length) : (i + 1),
```
Comment (josephbaylon) [inline e2e/utils/file.js:101]: Quick question, sort first weights are negative values right

Comment (saturninoabril) [reply e2e/utils/file.js:101]: Correct

File: e2e/run_test_cycle.js [hunk 2]
```diff
@@ -142,61 +166,69 @@ async function testLoop() {
...
+        tryNext: true,
+        count: newCount,
+        retry: 0,
+        message: 'Continue testing',
+    };
+}
 
-        return acc;
-    }, {});
+async function runSpec(count = 0, retry = 0) {
```
Comment (josephbaylon) [inline e2e/run_test_cycle.js:217]: Isn't this still recursion thru async/await operation?

**Review by josephbaylon — APPROVED** [review]
LGTM. Just a couple of questions

File: e2e/run_test_cycle.js (hunk 2 above)
Comment (saturninoabril) [reply e2e/run_test_cycle.js:217]: Right, still thru async/await, the main difference which I though where the issue is coming from is the [code here](https://github.com/mattermost/mattermost-webapp/pull/7913/files#diff-8686cf1e4a2ca1851e0b683287c8e22cde5d7c441ea88385df53c80893f16722L175):
```javascript
return testLoop(); // eslint-disable-line consistent-return
```
as compared to the [new approach here](https://github.com/mattermost/mattermost-webapp/pull/7913/files#diff-8686cf1e4a2ca1851e0b683287c8e22cde5d7c441ea88385df53c80893f16722R218):
```javascript
const fragment = await runSpecFragment(count, retry);
```

I maybe totally wrong but I tried couple of prod and unstable runs and CircleCI didn't complain/restart the container.

**Review by furqanmlk — APPROVED** [review]
Thanks @saturninoabril

**Pull request merged**